coderd/httpmw/loggermw/logger.go

```diff
@@ -79,25 +86,74 @@ func NewRequestLogger(log slog.Logger, message string, start time.Time) RequestL
 		written: false,
 		message: message,
 		start:   start,
+		actors:  make(map[rbac.SubjectType]rbac.Subject),
 	}
 }
 
 func (c *SlogRequestLogger) WithFields(fields ...slog.Field) {
 	c.log = c.log.With(fields...)
 }
 
+func (c *SlogRequestLogger) WithAuthContext(actor rbac.Subject) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.actors[actor.Type] = actor
+}
+
+func (c *SlogRequestLogger) addAuthContextFields() {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	usr, ok := c.actors[rbac.SubjectTypeUser]
+	if ok {
+		c.log = c.log.With(
+			slog.F("requestor_id", usr.ID),
+			slog.F("requestor_name", usr.FriendlyName),
+			slog.F("requestor_email", usr.Email),
+		)
+	} else if len(c.actors) > 0 {
+		for _, v := range c.actors {
+			c.log = c.log.With(
+				slog.F("requestor_name", v.FriendlyName),
+			)
+			break
```

Comment:
Unfortunately this picks a random actor. It'd be better if you had an order you tried to apply them in:
```go
// as a top level var probably
var actorLogOrder = []rbac.SubjectType{rbac.SubjectTypeUser,blah.Agent,blah.WorkspaceProxy,...}

// here:
for _, subjectType := range actorLogOrder {
  subj, ok := c.actors[subjectType]
  if !ok {
    continue
  }

  // add ID and name field
  // add email field if not empty
}